packages/jsii-rosetta/lib/jsii/assemblies.ts

```diff
@@ -171,36 +186,50 @@ export function allSnippetSources(assembly: spec.Assembly): AssemblySnippetSourc
   }
 }
 
-export function allTypeScriptSnippets(assemblies: readonly LoadedAssembly[], loose = false): TypeScriptSnippet[] {
-  const ret = new Array<TypeScriptSnippet>();
-
-  for (const { assembly, directory } of assemblies) {
-    const strict = enforcesStrictMode(assembly);
-    for (const source of allSnippetSources(assembly)) {
-      switch (source.type) {
-        case 'example':
-          // If an example is an infused example, we do not care about compiler errors.
-          // We are relying on the tablet cache to have this example stored already.
-          const [strictForExample, looseForExample] =
-            source.metadata?.infused !== undefined ? [false, true] : [strict, loose];
-          const location = { api: source.location, field: { field: 'example' } } as const;
-          const snippet = updateParameters(typeScriptSnippetFromSource(source.source, location, strictForExample), {
-            [SnippetParameters.$PROJECT_DIRECTORY]: directory,
-            ...source.metadata,
-          });
-          ret.push(fixturize(snippet, looseForExample));
-          break;
-        case 'markdown':
-          for (const snippet of extractTypescriptSnippetsFromMarkdown(source.markdown, source.location, strict)) {
-            const withDirectory = updateParameters(snippet, {
-              [SnippetParameters.$PROJECT_DIRECTORY]: directory,
-            });
-            ret.push(fixturize(withDirectory, loose));
-          }
-      }
-    }
-  }
-  return ret;
+export async function allTypeScriptSnippets(
+  assemblies: readonly LoadedAssembly[],
+  loose = false,
+): Promise<TypeScriptSnippet[]> {
+  return Promise.all(
+    assemblies
+      .flatMap((loaded) => allSnippetSources(loaded.assembly).map((source) => ({ source, loaded })))
+      .flatMap(({ source, loaded }) => {
+        switch (source.type) {
+          case 'example':
+            return [
+              {
+                snippet: updateParameters(
+                  typeScriptSnippetFromSource(
+                    source.source,
+                    { api: source.location, field: { field: 'example' } },
+                    isStrict(loaded),
+                  ),
+                  source.metadata ?? {},
+                ),
+                loaded,
+              },
+            ];
+          case 'markdown':
+            return extractTypescriptSnippetsFromMarkdown(source.markdown, source.location, isStrict(loaded)).map(
+              (snippet) => ({ snippet, loaded }),
+            );
+        }
+      })
+      .map(async ({ snippet, loaded }) => {
+        const isInfused = snippet.parameters?.infused !== undefined;
```

Comment:
Nit - I prefer `!= null`....

```suggestion
        const isInfused = snippet.parameters?.infused != null;
```